app/views/filters/_form.html.erb

```diff
@@ -42,9 +42,16 @@
           </span>
       </div>
 
-      <div id="granular_form" style="<%= !f.object.granular? ? 'display:none' : '' %>">
+      <div id="override_taxonomy_form" class="<%= f.object.allows_taxonomies_filtering? ? '' : 'hidden' %>">
+        <%= checkbox_f f, :override?,
+                       :help_inline => _("Filters inherit organizations and locations from their role by default. If override field is enabled, filter can override the set of its organizations and locations. Later role changes will not affect such filter. After disabling override field, role organizations and locations apply again."),
```

Comment:
It's fine if you want to put the popover next to the label itself too like in the Patternfly pattern. They do randomized tests of these patterns with users so I tend to trust their usefulness - besides, it promotes commonality across our projects (look how cockpit/freeipa/cloudforms implement help in forms).

About the locations/organizations, I think making it dependent on `Settings` is the way to go, through a helper or something. I'm aware it's a PITA but it won't make any sense to talk about organizations to an user that just has locations and viceversa